src/Service/AttachmentService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Attachment;
+use App\Entity\UserRequest;
+use App\Traits\UserAwareTrait;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Uid\Uuid;
+
+class AttachmentService
+{
+    use UserAwareTrait;
+
+    public function __construct(
+        private readonly EntityManagerInterface $em,
+        private readonly Security $security,
+        private readonly string $uploadsDirectory,
+    ) {
+    }
+
+    public function addAttachment(UserRequest $userRequest, mixed $file): void
+    {
+        if (!($file instanceof UploadedFile)) {
+            return;
+        }
+
+        $name = Uuid::v4().'.'.$file->guessExtension();
```

Comment:
Could it be some exceptions around here, and edge cases regarding the guessExtension method